firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/PatchMutation.java

```diff
@@ -142,6 +143,32 @@ public void applyToLocalView(MutableDocument document, Timestamp localWriteTime)
         .setHasLocalMutations();
   }
 
+  @Override
+  public Mutation squash(
+      Mutation baseMutation, MutableDocument document, Timestamp localWriteTime) {
+    if (!getPrecondition().isValidFor(document)) {
+      return baseMutation;
+    }
+
+    Map<FieldPath, Value> transformResults = localTransformResults(localWriteTime, document);
```

Comment:
Same here, transforms need to be applied on `document` + `baseMutation`.

Reply:
I actually think we need to do two different things:

- First round, we use the document
- Second round, we use the base mutation

So ideally we would change `localTransformResults` to take an ObjectValue instead of the document.